clang/unittests/Analysis/FlowSensitive/DataflowEnvironmentTest.cpp

```diff
@@ -403,4 +405,35 @@ TEST_F(EnvironmentTest,
               Contains(Member));
 }
 
+TEST_F(EnvironmentTest, Stmt) {
+  using namespace ast_matchers;
+
+  std::string Code = R"cc(
+      struct S {int i;};
+      void foo() {
+        S AnS = S{1};
+      }
```

Comment:
A variable declaration outside a function doesn't actually exist within a Stmt, nor is it a Stmt.

In order to analyze a global variable declaration, one option is to synthesize a DeclStmt that holds the VarDecl. I was trying to keep this test simple, but could add an additional test for this synthesis pattern?

Reply:
> A variable declaration outside a function doesn't actually exist within a Stmt, nor is it a Stmt.

But isn't this the use case for which you're making the change in this PR?

I don't see why you need the variable declaration to be a `Stmt` though. In any case, you're only analyzing the initializer, not the `DeclStmt` itself. Instead of searching for the `DeclStmt`, you could search for the `VarDecl` and then initialize the `Environment` with the `getInit()` of that `VarDecl`. This wouldn't make the test any more complicated and would be equivalent to what you're testing here?

Reply:
Having looked at this again, I now realize that you're analyzing the `DeclStmt` itself, and not just the declaration's initializer. I think analyzing just the initializer would have the same effect for the purposes of this test -- but either of these two options seems fine, so feel free to leave this as is.